### ISSUE TITLE
no-undef false positive with es6 destructuring and defaults.

### ISSUE THREAD

**Issue by mythmon** [eslint/eslint#2214]
With this code

```javascript
function foo() {
  var [a, b=4] = [1, 2];
  return {a, b};
}
```

and this .eslintrc

```json
{
  "ecmaFeatures": {
    "destructuring": true
  }
}
```

I get the error `3:13  error  'b' is not defined  no-undef`. If the default value for `b` is removed, the error doesn't happen.

**Comment by nzakas** [comment]
Might be an escope issue, will have to take a look.

**Comment by pola88** [comment]
Hey guys,

Something new about this? I have the same problem but I add the default values in the function params.

Example:
```
function(options = {}) {
....
}

Thanks,

**Comment by nzakas** [comment]
Working on this.

**Comment by nzakas** [comment]
This appears to have been fixed with the latest escope changes. Just adding a test to watch for regressions in the future.
